Force slaptest to be quiet when no problem is found. Use -v to display "success" message.
Patch sent upstream (ITS#4930: slaptest should be quiet on success)

--- openldap-2.3.34/servers/slapd/slaptest.c.orig	2007-01-02 22:43:59.000000000 +0100
+++ openldap-2.3.34/servers/slapd/slaptest.c	2007-04-17 10:54:36.000000000 +0200
@@ -103,7 +103,8 @@
 		}
 	}
 
-	fprintf( stderr, "config file testing succeeded\n");
+	if (verbose)
+		fprintf( stderr, "config file testing succeeded\n");
 
 	slap_tool_destroy();
 
